### DISCUSSION TITLE
I2C support?

### DISCUSSION THREAD

**Discussion by phil-barrett** [witnessmenow/ESP32-Cheap-Yellow-Display#32] [General]
I went looking for your discord channel but couldn’t find it. Discover shows nothing named cheap-yellow-display. Not a discord pro for sure but gave up after a few minutes of trying.

So, I’ll ask the question here, does the CYD have an I2C interface? I think it might be a nice peripheral to a family of devices I’m working with.

**Comment by peasdesign** [comment]
My cyd is esp32-2432s024N.
I2C is not functional, but the 22-pin is alive. The adjacent 35-pin for the temperature and humidity sensor can also be used, but it is input-only. I've searched for another decent pin from other connectors but couldn't find any.

**Reply by peasdesign** [reply]
No response.
 https://playground.arduino.cc/Main/I2cScanner/

**Reply by phil-barrett** [reply]
Thanks.  I think the I2C controller needs to be set up in the firmware on the CYD.  What I'd like is a device that I can send specific data to via an I2C connection.  The FW on the BYD would display this data. Think auxiliary data display for a milling machine, for example.  It might be possible to use a UART.  However, after looking at the CYD a bit more, I think there are other solutions that might be easier for me. In particular, I like the idea of an RP2040 based device.

**Comment by peasdesign** [comment]
I am currently consumed by a reckless idea. The idea is to completely remove resistors R18 and R25. This might disable the touch controller, but I am originally using the 'N' variant, meaning a CYD without a touch controller. I believe these measures will free up the IO21 pin and allow the connector to be used as an I2C interface

**Comment by witnessmenow** [comment]
I've updated some of the documentation yesterday, including adding a link to the discord on the main readme.

The easiest way to connect an i2c device is with the CN1 connector, the one nearest the micro SD card slot. It has two free gpio pins, 3.3v and ground.

I've added some instructions around how I connected the nunchuck, any other i2c device should be the same

https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/blob/main/ADDONS.md#wiring

The only other thing that you need to do is specify the pins when starting the i2c wire interface

https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/blob/155f70e485994c9423c3af1035bb5ce72899ff01/Examples/InputTests/NunchuckTest/NunchuckTest.ino#L78

**Comment by DB59000** [comment]
If you use some librairy and instance object  for instance like this

#include <WEMOS_SHT3X.h>
SHT3X sht30(0x45);

#include "Adafruit_TCS34725.h"
Adafruit_TCS34725 tcs = Adafruit_TCS34725(TCS34725_INTEGRATIONTIME_614MS, TCS34725_GAIN_1X);

#include <VL53L1X.h>
VL53L1X sensor;

you need to insert this into your setup:

  // use CN1 for I2C (SDA yellow on  GPIO27, SCL blue on GPIO22)
  Wire.end();
  Wire.setPins(27, 22);
  Wire.begin();
